docs/sdk/src/guides/enable_elastic_scaling.rs

```diff
@@ -0,0 +1,249 @@
+//!    This is not in the scope of the relay chain functionality. Parachains can already start
+//!    implementing such autoscaling, but we aim to provide a framework/examples for developing
+//!    autoscaling strategies.
+//!    Tracked by [this issue](https://github.com/paritytech/polkadot-sdk/issues/1487).
+//!    An in-progress external implementation by RegionX can be found [here](https://github.com/RegionX-Labs/On-Demand).
+//!
+//! Another important constraint is that when a parachain forks, the throughput decreases and
+//! latency increases because the number of blocks backed per relay chain block goes down.
+//!
+//! ## Using elastic scaling
+//!
+//! [Here](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/zombienet-sdk-tests/tests/elastic_scaling/slot_based_12cores.rs)
+//! is a zombienet test which exercises 500ms parachain blocks, which you can modify to test your
+//! parachain after going through the required phases below.
+//!
+//! ### Prerequisites
+//!
+//! - Ensure Asynchronous Backing (6-second blocks) has been enabled on the parachain using
+//!   [`crate::guides::async_backing_guide`].
+//! - Ensure the `AsyncBackingParams.max_candidate_depth` value is configured to a value that is at
+//!   least double the maximum targeted parachain velocity. For example, if the parachain will build
+//!   at most 3 candidates per relay chain block, the `max_candidate_depth` should be at least 6.
+//! - Ensure enough coretime is assigned to the parachain.
+//! - Ensure the `CandidateReceiptV2` node feature is enabled on the relay chain configuration (node
+//!   feature bit number 3).
+//!
+//! <div class="warning">Phase 1 is NOT needed if using the <code>polkadot-parachain</code> or
+//! <code>polkadot-omni-node</code> binary, or <code>polkadot-omni-node-lib</code> built from the
+//! latest polkadot-sdk release! Simply pass the <code>--experimental-use-slot-based</code>
+//! ([`polkadot_omni_node_lib::cli::Cli::experimental_use_slot_based`]) parameter to the command
+//! line and jump to Phase 2.</div>
+//!
+//! ### Phase 1 - (For custom parachain node) Update Parachain Node
+//!
+//! This assumes you are using
+//! [the latest parachain template](https://github.com/paritytech/polkadot-sdk/tree/master/templates/parachain).
+//!
+//! This phase consists of plugging in the new slot-based collator, which is a requirement for
+//! elastic scaling.
+//!
+//! 1. In `node/src/service.rs` import the slot based collator instead of the lookahead collator, as
+//!    well as the `SlotBasedBlockImport` and `SlotBasedBlockImportHandle`.
+#![doc = docify::embed!("../../cumulus/polkadot-omni-node/lib/src/nodes/aura.rs", slot_based_colator_import)]
+//!
+//! 2. Modify the `ParachainBlockImport` and `Service` type definitions:
+//! ```ignore
+//! type ParachainBlockImport = TParachainBlockImport<
+//! 	    Block,
+//! 	    SlotBasedBlockImport<Block, Arc<ParachainClient>, ParachainClient>,
```

Comment:
I think we don't need `SlotBasedBlockImport` anymore if we only support 6s slots or more. wdyt @bkchr